src/python/pants/rules/core/list_target_types.py

```diff
@@ -32,56 +34,87 @@ class TargetTypes(Goal):
-    target_type: Type[Target], console: Console, *, longest_target_alias: int
-) -> str:
-    """Return a single line description of the target type."""
-    formatted_alias = console.cyan(f"{target_type.alias:>{longest_target_alias + 1}}:")
-    description = get_docstring_summary(target_type) or "<no description>"
-    return f"{formatted_alias} {description}"
+@dataclass(frozen=True)
+class AbbreviatedTargetInfo:
+    alias: str
+    description: Optional[str]
 
+    @classmethod
+    def create(cls, target_type: Type[Target]) -> "AbbreviatedTargetInfo":
+        return cls(alias=target_type.alias, description=get_docstring_summary(target_type))
+
+    def format_for_cli(self, *, console: Console, longest_target_alias: int) -> str:
+        formatted_alias = console.cyan(f"{self.alias:>{longest_target_alias + 1}}:")
+        description = self.description or "<no description>"
+        return f"{formatted_alias} {description}"
 
-def verbose_target_information(
-    target_type: Type[Target], console: Console, union_membership: UnionMembership
-) -> str:
-    """Return a multiline description of the target type and all of its fields."""
-    target_type_description = get_docstring(target_type)
-    if target_type_description:
-        target_type_description = console.blue(target_type_description)
 
-    fields = target_type.class_field_types(union_membership=union_membership)
-    longest_field_name = max(len(field.alias) for field in fields)
+@dataclass(frozen=True)
+class FieldInfo:
+    alias: str
+    description: Optional[str]
+    type_hint: str
+    default: str
 
-    # TODO: consider hard wrapping fields. It's confusing to read when fields soft wrap.
-    def format_field(field: Type[Field]) -> str:
-        field_alias_text = f"  {field.alias}"
-        field_alias = console.cyan(f"{field_alias_text:<{longest_field_name + 2}}")
-        description = get_docstring(field, flatten=True) or ""
+    @classmethod
+    def create(cls, field: Type[Field]) -> "FieldInfo":
+        description = get_docstring(field, flatten=True)
         if issubclass(field, PrimitiveField):
             raw_value_type = get_type_hints(field.compute_value)["raw_value"]
         elif issubclass(field, AsyncField):
             raw_value_type = get_type_hints(field.sanitize_raw_value)["raw_value"]
         else:
             raw_value_type = get_type_hints(field.__init__)["raw_value"]
-        type_info = console.green(
-            f"(type: {pretty_print_type_hint(raw_value_type)}, default: {field.default})"
+        type_hint = pretty_print_type_hint(raw_value_type)
+        return cls(
+            alias=field.alias,
+            description=description,
+            type_hint=type_hint,
+            default=str(field.default),
+        )
+
+    def format_for_cli(self, *, console: Console) -> str:
+        field_alias = console.magenta(f"{self.alias}")
+        type_info = console.cyan(f"  type: {self.type_hint}, default: {self.default}")
```

Comment:
I used the same color scheme as `./pants help` for consistency:

![Screen Shot 2020-03-26 at 10 19 48 PM](https://user-images.githubusercontent.com/14852634/77724322-ea5aca80-6faf-11ea-808a-57a038192aa6.png)